controller/src/controller/models/domain/rsptag.py

```diff
@@ -106,6 +107,38 @@ class RSPImageType(Enum):
 ]
 
 
+def _calculate_date(tagdata: dict[str, str]) -> datetime.datetime | None:
+    """Calculate the date when the image should have been created.
+
+    Parameters
+    ----------
+    tagdata
+        The match groups from the regular expression tag match.
+
+    Returns
+    -------
+    datetime.datetime | None
+        The image creation date if it can be gleaned from the tag.
+    """
+    year = tagdata.get("year")
+    if not year:
+        return None
+    week = tagdata.get("week")
+    if year and week:
+        jan1 = datetime.datetime(int(year), 1, 1, tzinfo=datetime.UTC)
+        # This isn't exact, it's not isoformat, and it's not pretty, but it is
+        # also close enough for our purposes.
+        # We start counting at week one in our tags.
+        return jan1 + datetime.timedelta(weeks=int(week) - 1)
```

Comment:
I think this is a complicated way of writing `datetime.fromisocalendar(int(year), int(week), 4)`. You do unfortunately have to specify the day of the week, but all of this is specific to Rubin practices anyway so we should just set whatever day of the week it is that we tag weeklies. The above is Friday; I'm not sure if that's right.

Reply:
Almost.  The above is Thursday (which is the day we build).  But yes, fromisocalendar() is what I wanted (and forgot existed)

Reply:
Although, irritatingly, that returns a naive object.